doc/proposals/cli-ux-phase1.md

```diff
@@ -0,0 +1,121 @@
+# User Experience CLI Improvements | Phase 1
+
+Implementation Owner: @joelanford
+
+Status: Draft
+
+[Background](#Background)
+
+[Goal](#Goal)
+
+[Use cases](#Use_cases)
+
+[Proposed CLI commands](#Proposed_CLI_commands)
+
+[References](#References)
+
+## Background
+
+The SDK CLI is one of the primary tools in the Operator Framework for operator developers and
+deployers. However, the current user experience (UX) leaves a lot to be desired. Currently, the
+tools and documentation that help a user create, develop, test, package, run, and publish an
+operator are spread among many different repositories, making for a steep learning curve for 
+newcomers.
+
+## Goal
+
+Describe an initial set of CLI additions to support a simplified workflow for an Operator SDK user to run an Operator SDK-developed project in a cluster using OLM.
+
+
+## Use cases
+
+1. As an operator developer, I want to deploy OLM to a development cluster.
+2. As an operator developer, I want to create the resources necessary for running my operator via OLM.
+3. As an operator developer, I want to run my operator via OLM.
+
+## Proposed CLI commands
+
+### `operator-sdk alpha`
+
+Many other Kubernetes CLI projects (e.g. `kubebuilder`, `kubectl`, and `kubeadm`) have an `alpha` subcommand tree that is used to introduce new functionality to the CLI while giving users a cue that the functionality is subject to change. This gives maintainers flexibility while iterating on new CLI features.
+
+The general idea is that new subcommand features should be introduced under the `alpha` subcommand (e.g. `operator-sdk alpha my-command`) and then moved to their own top-level commands once the implementation has matured (e.g. `operator-sdk my-command`).
+
+### `operator-sdk alpha olm init`
+
+Deploys the Operator Lifecycle Manager (OLM). The initial proof-of-concept will initialize OLM using manifests from OLM's repository, using the following URL format, based on provided flags:
+
+`https://raw.githubusercontent.com/operator-framework/operator-lifecycle-manager/master/deploy/<cluster-type>/manifests/<olm-version>/`
+
+#### Flags for `olm init`
+
+| Flag             | Type   | Description                                                                                     |
+|------------------|--------|-------------------------------------------------------------------------------------------------|
+| `--cluster-type` | string | The type of cluster to deploy OLM to, one of `upstream`, `ocp`, `okd` (default `upstream`)      |
+| `--olm-version`  | string | The version of OLM to deploy (default `latest`)                                                 |
+| `--kubeconfig`   | string | Path for custom Kubernetes client config file (overrides the default locations)                 |
+
+**Open questions:** 
+1. Do we need to support different cluster types? Or should we just always use `upstream` since OLM is already installed by default in OCP and OKD clusters?
```

Comment:
Is it installed in older versions of olm and okd as well? We could still leave support for those and verify that it's not installed already.

Reply:
The different deploy directories for each cluster type have their [values.yaml](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/deploy/upstream/values.yaml#L4-L5) hardcoded on where to deploy OLM and what namespaces OLM watches for OLM resources.

If a user is sharing an upstream/ocp/okd cluster with OLM already installed, is it useful for them to setup another instance of OLM for their testing purposes?
If so then we'll probably want more optional flags or a config file for `olm init` to override the default values.yaml on where to install OLM.
https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/install/install.md#building-deployment-resources-for-any-cluster

Reply:
But we probably don't need to support this to begin with until the use case for multiple OLM installations is more clear. 
Plus seems like we've agreed to rely on the default installations of OLM in ocp/okd going forward.